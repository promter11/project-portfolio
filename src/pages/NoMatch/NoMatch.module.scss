@import "../../styles/variables";

.noMatch {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100%;
  text-align: center;

  .title {
    max-width: 47rem;
    margin: 0 auto 2rem auto;
    font-size: 7.8rem;
    font-weight: 900;
    line-height: 125%;
    text-transform: capitalize;
    color: $black-color;
  }

  .text {
    max-width: 52rem;
    margin: 0 auto 3rem auto;
    font-size: 1.8rem;
    font-weight: 300;
    line-height: 150%;
  }

  .link {
    display: inline-block;
    font-size: 1.4rem;
    font-weight: bold;
    line-height: 1;
    text-transform: uppercase;
    padding: 1.5rem 3rem;
    border: 0.2rem solid $grey-text-color;
    box-shadow: rgba(240, 46, 170, 0.4) 0.5rem 0.5rem,
      rgba(240, 46, 170, 0.3) 1rem 1rem, rgba(240, 46, 170, 0.2) 1.5rem 1.5rem,
      rgba(240, 46, 170, 0.1) 2rem 2rem, rgba(240, 46, 170, 0.05) 2.5rem 2.5rem;
    transition: all 0.25s ease-in-out;
    background-color: transparent;
    color: $grey-text-color;

    &:hover {
      border: 0.2rem solid transparent;
      background-color: rgba(240, 46, 170, 1);
      color: $white-color;
    }
  }
}
