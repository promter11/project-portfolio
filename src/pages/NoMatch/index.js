import React from "react";
import { Link } from "react-router-dom";

import Heading from "../../components/Heading";
import Container from "../../components/Container";

import { noMatchText } from "../../common/jsonText/noMatchText";
import { loadState } from "../../common/localStorage";

import styles from "./NoMatch.module.scss";

const NoMatch = () => {
  const {
    LanguageReducer: {
      defaultLanguage: { language },
    },
  } = loadState();

  return (
    <section className={styles.noMatch}>
      <Container>
        <Heading className={styles.title} level={1}>
          {noMatchText[language].title}
        </Heading>
        <p className={styles.text}>{noMatchText[language].desc}</p>
        <Link className={styles.link} to={noMatchText[language].link.path}>
          {noMatchText[language].link.text}
        </Link>
      </Container>
    </section>
  );
};

export default NoMatch;
