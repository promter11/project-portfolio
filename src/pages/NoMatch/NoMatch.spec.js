import React from "react";
import { shallow } from "enzyme";

import NoMatch from "./index";

const props = {
  language: "RU",
};

describe("NoMatch page", () => {
  it("renders correctly with props", () => {
    const wrapper = shallow(<NoMatch {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
