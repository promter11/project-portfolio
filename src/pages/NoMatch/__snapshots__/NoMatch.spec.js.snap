// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoMatch page renders correctly with props 1`] = `
<section
  className="noMatch"
>
  <Container>
    <Heading
      className="title"
      level={1}
    >
      Страница не найдена
    </Heading>
    <p
      className="text"
    >
      Страница, которую вы искали, не может быть найдена. Она могла быть удалена, переименована или вообще не существовать.
    </p>
    <Link
      className="link"
      to="/"
    >
      Главная страница
    </Link>
  </Container>
</section>
`;
